keps/sig-api-machinery/0015-dry-run.md

```diff
@@ -110,9 +110,33 @@ type AdmissionRequest struct {
 }
 ```
 
+## Generated values
+
+Some values of the object are typically generated before the object is persisted:
+- generateName can be used to assign a unique random name to the object,
+- creationTimeStamp records the time of creation,
+- UID uniquely identifies the object and is randomly generated (non-deterministic),
+- resourceVersion tracks the persisted version of the object.
+
+Most of these values are not useful in the context of dry-run, and could create
+some confusion. The UID and the generated name would have a different value in a
+dry-run and non-dry-run creation. These values will be left empty when
+performing a dry-run.
+
+CreationTimeStamp is also generated on creation, but there are less ways to
```

Comment:
you think this will be used as an expensive "get the current server time" API?

Reply:
> you think this will be used as an expensive "get the current server time" API?

Delete is weird in general.  I think an exploration of deletion semantics will be required as we try to make it useable.

About creationtimestamp, as a consumer, wouldn't you do it to set the skew?  We've had requests for similar things.  It wouldn't stop me allowing it to be set if that were easier though.